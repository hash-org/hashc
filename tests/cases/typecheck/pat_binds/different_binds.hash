// run=fail, stage=typecheck


main := () => {
    k := (1, 2);

    match k {
        (mut a, b) | (a, b) => {}
    }
}
