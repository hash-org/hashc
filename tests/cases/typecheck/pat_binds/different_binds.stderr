error[0079]: variable `a` is bound inconsistently across or-patterns
 --> $DIR/different_binds.hash:8:14
7 |       match k {
8 |           (mut a, b) | (a, b) => {}
  |                ^ first binding of variable
9 |       }

 --> $DIR/different_binds.hash:8:23
7 |       match k {
8 |           (mut a, b) | (a, b) => {}
  |                         ^ bound differently
9 |       }
